de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,343 @@
+		sei.setName("Testsein");
+		assertFalse("Not the same name", importer.checkIfCorrectSei(sei, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongUUID() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setType(sei.getType());
+		assertFalse("Not the same uuid", importer.checkIfCorrectSei(sei2, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongType() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setUuid(sei.getUuid());
+		assertFalse("Not the same type", importer.checkIfCorrectSei(sei2, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongNumberOfChildren() {
+		TestStructuralElementOther tsei2 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setParent(sei);
+		assertFalse("Not the same number of children", importer.checkIfCorrectSei(sei, mat));
+	}
+
+	@Test
+	public void testCheckIfCorrectSeiWrongChildren() {
+		TestStructuralElement tsei2 = new TestStructuralElement(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setParent(sei);
+		mat = exporter.exportSei(sei);
+		TestStructuralElementOther tsei3 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei3 = tsei3.getStructuralElementInstance();
+		sei3.setName("testseiChild");
+		TestStructuralElementOther tsei4 = new TestStructuralElementOther(testConcept);
+		StructuralElementInstance sei4 = tsei4.getStructuralElementInstance();
+		sei4.setName("testseiChild2");
+		sei2.setParent(sei3);
+		sei4.setParent(sei);
+		assertFalse("Not the same children", importer.checkIfCorrectSei(sei, mat));
+	}
+
+	@Test
+	public void testImportCasDeleteCas() throws IOException {
+		TestCategoryAllProperty tc2 = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc2);
+		mat = exporter.exportSei(sei);
+		TestCategoryComposition tc1 = new TestCategoryComposition(testConcept);
+		tsei.add(tc1);
+		assertTrue("Sei has two CategoryAssinments", sei.getCategoryAssignments().size() == 2);
+		importer.importSei(sei, mat);
+		assertTrue("Only one CategoryAssinment", sei.getCategoryAssignments().size() == 1);
+		assertTrue("Right CategoryAssinment included",
+				sei.getCategoryAssignments().get(0).getName().equals(tc2.getName()));
+	}
+
+	@Test
+	public void testImportAPIDeleteAPI() throws IOException {
+		EReferenceTest tc2 = new EReferenceTest(testConcept);
+		tsei.add(tc2);
+		TestCategoryAllProperty tc3 = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc3);
+		APropertyInstance nInstance = sei.getCategoryAssignments().get(0).getPropertyInstances().get(0);

```

Comment:
It's reletively hard to follow this test as someone from the outside... All this index handling and the connection of tsei and sei are hard to understand... Could you add som comments in your tests?